ssl/statem/statem_clnt.c

```diff
@@ -437,10 +440,27 @@ static WRITE_TRAN ossl_statem_client13_write_transition(SSL *s)
 
     case TLS_ST_CR_CERT_REQ:
         if (s->post_handshake_auth == SSL_PHA_REQUESTED) {
-            if (s->ext.compress_certificate_rx == TLSEXT_comp_cert_none)
-                st->hand_state = TLS_ST_CW_CERT;
+#ifndef OPENSSL_NO_COMP
+            if (s->options & SSL_OP_NO_CERTIFICATE_COMPRESSION ||
+                    s->ext.compress_certificate_rx == TLSEXT_comp_cert_none)
+                ret = 0;
+            else if (s->cert_comp_cb != NULL)
+                ret = s->cert_comp_cb(s, s->ext.compress_certificate_rx, s->ctx->cert_comp_arg);
```

Comment:
Ouch. I really don't like having side effects in the transition function. The transition function should be as simple as possible.

Reply:
I'll look into updating these side-effects issues. Takes more time than some of the other stuff.